<launch>

    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <param name="m" type="double" value="1.98" />
    <param name="I_z" type="double" value="0.03" />
        
    <!-- Track parameters -->
    <param name="trackShape" type="str" value="oval" />

   
    <!-- VISUALIZATION -->
    <node pkg="barc" type="plotCarTrajectory.py" name="visualization" output="screen">
        <param name="plotGPS" type="bool" value="true" />
    </node>


    <!-- SIMULATOR -->
    <node pkg="barc" type="vehicleSimulator.py" name="simulator" output="screen">
        <!-- sensor freq update -->
        <param name="gps_freq_update" type="double" value="4.0" />
        <!-- delay in the simulator -->
        <param name="delay_df" type="double" value="0.0" />
        <param name="delay_a" type="double" value="0.0" />
        <!-- simulator model parameters -->
        <param name="c_f" type="double" value="0.05" />
        <param name="dt" type="double" value="0.01" />
        <!-- Tire model parameters -->
        <param name="B" type="double" value="6.0" />
        <param name="C" type="double" value="1.6" />
        <param name="mu" type="double" value="0.8" />
        <param name="g" type="double" value="9.81" />
        <!-- simulator sensor noise -->
        <param name="n_bound" type="double" value="1.5" />
        <param name="x_std" type="double" value="0.02" />
        <param name="y_std" type="double" value="0.02" />    
        <param name="v_std" type="double" value="0.0" />    
        <param name="ax_std" type="double" value="1.0" />
        <param name="ay_std" type="double" value="1.0" />
        <param name="psiDot_std" type="double" value="0.0001" />
    </node>
    
    <!--DYNAMIC OBSERVER-->  
    <!--<node pkg="barc" type="stateEstimatorClean.py" name="state_estimator" output="screen">-->
    <!--<node pkg="barc" type="stateEstimatorMultiRate.py" name="state_estimator" output="screen">-->
    <node pkg="barc" type="stateEstimatorPsiInput.py" name="state_estimator" output="screen">
    <!--<node pkg="barc" type="stateEstimatorInputs.py" name="state_estimator" output="screen">-->
    <!-- delay in state estuiamtor -->
        <param name="delay_df" type="double" value="0.0" />
        <param name="delay_a" type="double" value="0.0" />
        <!-- ekf properties -->
         <param name="thReset" type="double" value="0.5" />
         <param name="vSwitch" type="double" value="1.3" />
         <param name="psiSwitch" type="double" value="1.2" />

         <param name="Qx_hs" type="double" value="0.01" />
         <param name="Qx_ls" type="double" value="0.01" />
     
         <param name="Qy_hs" type="double" value="0.01" />
         <param name="Qy_ls" type="double" value="0.01" />
     
         <param name="Qvx_hs" type="double" value="0.5" />
         <param name="Qvx_ls" type="double" value="0.5" />
     
         <param name="Qvy_hs" type="double" value="0.5" />
         <param name="Qvy_ls" type="double" value="0.5" />
     
         <param name="Qax_hs" type="double" value="1.0" />
         <param name="Qax_ls" type="double" value="1.0" />
     
         <param name="Qay_hs" type="double" value="10.0" />
         <param name="Qay_ls" type="double" value="1.0" />
     
         <param name="Qpsi_hs" type="double" value="0.1" />
         <param name="Qpsi_ls" type="double" value="10.0" />
     
         <param name="QpsiDot_hs" type="double" value="1.0" />
         <param name="QpsiDot_ls" type="double" value="0.01" />
     
         <param name="Rx_hs" type="double" value="1.0" />
         <param name="Rx_ls" type="double" value="40.0" />
     
         <param name="Ry_hs" type="double" value="1.0" />
         <param name="Ry_ls" type="double" value="40.0" />
     
         <param name="Rvx_hs" type="double" value="0.1" />
         <param name="Rvx_ls" type="double" value="0.1" />
     
         <param name="Rax_hs" type="double" value="10.0" />
         <param name="Rax_ls" type="double" value="10.0" />
     
         <param name="Ray_hs" type="double" value="30.0" />
         <param name="Ray_ls" type="double" value="10.0" />
    
         <param name="RpsiDot_hs" type="double" value="0.1" />
         <param name="RpsiDot_ls" type="double" value="0.1" />
    
         <param name="Rvy_hs" type="double" value="0.01" />
         <param name="Rvy_ls" type="double" value="0.01" />
    </node>

    <!-- LOW LEVEL CONTROLLER -->
    <!--<node pkg="barc" type="controller_low_level.py" name="low_level_controller" /> -->

     <!-- CONTROL -->
    <node pkg="barc" type="controllerMain.py" name="control" output="screen">
        <param name="mode" type="str" value="simulations" />
        <param name="saveData" type="bool" value="true" />
    </node>


</launch>
